site.yml

```diff
@@ -209,24 +209,30 @@
     - role: haproxy
       haproxy_type: swift
       tags: ['openstack', 'swift', 'control', 'haproxy', 'infra']
+      when: swift.enabled|bool
 
     - role: swift-object
       tags: ['openstack', 'swift', 'data']
+      when: swift.enabled|bool
 
     - role: swift-account
       tags: ['openstack', 'swift', 'data']
+      when: swift.enabled|bool
 
     - role: swift-container
       tags: ['openstack', 'swift', 'data']
+      when: swift.enabled|bool
 
     - role: swift-proxy
       tags: ['openstack', 'swift', 'control']
+      when: swift.enabled|bool
 
 - name: swift bootstrap rings
   hosts: swiftnode_primary
   roles:
     - role: swift-ring
       tags: ['openstack', 'swift', 'swift-ring', 'data']
+      when: swift.enabled|bool
```

Comment:
Same here, why would there be a host in swiftnode_primary if swift is not enabled.